tasks/init.rb

```diff
@@ -5,7 +5,8 @@
 require 'open3'
 
 def get(fact)
-  cmd_string = "facter -p #{fact}"
+  facter = (%r{mingw} =~ RUBY_PLATFORM) ? '"C:\Program Files\Puppet Labs\Puppet\bin\facter"' : '/opt/puppetlabs/puppet/bin/facter'
+  cmd_string = "#{facter} -p #{fact}"
```

Comment:
per @garethr 's [comment](https://tickets.puppetlabs.com/browse/MODULES-5747), wouldn't it be better to use the Ruby interface and replace the shell command altogether?

Reply:
Yeah, that seems like the best option.

Reply:
That was what my initial commit on this PR did.  My understanding was that the ruby interface did not provide access to Puppet facts, which would break https://github.com/puppetlabs/puppetlabs-facter_task/commit/56fce4a67298e950e3484df18a78ff30000410f4, so I changed it to this.

Is it possible to pull Puppet facts using the Ruby facter interface?

Reply:
Ah right. We should probably extend that, but it doesn't exist yet.

Reply:
That's what I was afraid of.  It's a bummer, since that does make the code much cleaner.

Reply:
ah ok, my bad. carry on.